catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -233,6 +233,12 @@ public DeleteStorageResponse delete(DeleteStorageRequest deleteRequest)
                 // For deletion we can ignore the qualifier and assume everything under a given ID is
                 // to be removed.
                 Path contentIdDir = getContentItemDir(new URI(deletedContentItem.getUri()));
+
+                if (contentIdDir == null) {
+                    throw new StorageException(
```

Comment:
We shouldn't throw an exception here as it will skip all the other metacards in the delete request.